mallet dies with SIGSEGV

Hope you're doing well and you remember "pf tunnel" depends on mallet. Recently I tried to use it on my newly deployed EndeavourOS and found that mallet wouldn't work on it.

```
~ [☁️  kmizumar@preferred.jp]
❯ pf tunnel status         
● ssh
 Status: running
    PID: 193205
 Stderr: (/tmp/pfutil-tunnel/ssh.stderr.log)
2022-12-12T18:16:26+09:00 INF internal.go:32] Starting ssh...

● mallet
 Status: not running
 Stderr: (/tmp/pfutil-tunnel/mallet.stderr.log)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x7e0ef6]

goroutine 65 [running]:
github.com/ryotarai/mallet/pkg/proxy.(*Proxy).handleConn(0xc00002a200, 0xc00039c000, 0x0, 0x0)
	/home/runner/work/mallet/mallet/pkg/proxy/proxy.go:97 +0x116
github.com/ryotarai/mallet/pkg/proxy.(*Proxy).Start.func1(0xc00002a200, 0xc00039c000)
	/home/runner/work/mallet/mallet/pkg/proxy/proxy.go:74 +0x39
created by github.com/ryotarai/mallet/pkg/proxy.(*Proxy).Start
	/home/runner/work/mallet/mallet/pkg/proxy/proxy.go:73 +0x4dd
```

This panic happened at https://github.com/ryotarai/mallet/blob/master/pkg/proxy/proxy.go#L97 and caused by nil pointer dereference. Noticed that `raddr` (RemoteAddr()) of `conn` object was nil on my box.

```
	p.Logger.Debug().Str("src", conn.RemoteAddr().String()).Str("dst", dest).Msg("Starting proxy")
```

Not sure about why this issue won't happen on other environments, including my previous Ubuntu 22.04LTS. 

----

As a quick fix, I changed RemoteAddr to LocalAddr since `conn.laddr` is always non-nil and I don't care about the log messages that's not emitted.

--- a/pkg/proxy/proxy.go
+++ b/pkg/proxy/proxy.go
@@ -94,7 +94,11 @@ func (p *Proxy) handleConn(conn *net.TCPConn) error {
 	}
 	conn = newConn
 
-	p.Logger.Debug().Str("src", conn.RemoteAddr().String()).Str("dst", dest).Msg("Starting proxy")
+	srcAddr := ""
+	if addr := conn.RemoteAddr(); addr != nil {
+		srcAddr = addr.String()
+	}
+	p.Logger.Debug().Str("src", srcAddr).Str("dst", dest).Msg("Starting proxy")
 
 	// me -> chisel
 	pipeOutR, pipeOutW := io.Pipe()

